### DISCUSSION TITLE
Blog comment on Distance Metrics in Vector Search

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#1794] [Docs/blog comments]
# Blog comment on Distance Metrics in Vector Search

Learn about why you need distance metrics in vector search and the metrics implemented in Weaviate (Cosine, Dot Product, L2-Squared, Manhattan, and Hamming).

https://weaviate.io/blog/distance-metrics-in-vector-search

**Comment by Vodyanoy17** [comment]
Dot Produc:"It keeps the intuition that a smaller distance value means they are similar." is not true
A*B= 13, but A*A= 16+1=17 ,

**Reply by patmejia** [reply]
## Weaviate's Distance Metrics $-(\vec{A} \cdot \vec{B})$ vs. Traditional Dot Product's Similarity Measurement:

> Negative of the dot product: "It keeps the intuition that a smaller distance value means they are similar." - Weaviate's Docs is correct within the context of Weaviate's distance metrics or vector search.

### 1. Euclidean Geometry:

Similarity Measurement: In Euclidean geometry, the dot product measures the ***alignment*** and magnitude of vectors, with higher values indicating greater similarity. Larger magnitudes increase the dot product when vectors are aligned or at an acute angle, but larger angles, especially over $> 90^\circ$, can result in low or negative values regardless of magnitude.

###### Example: <br>For vectors $\vec{A} = [4, 0, 1]$ and $\vec{B} = [3, 0, 1]$, <br>the dot product $\vec{A} \cdot \vec{B} = 13$ suggests moderate similarity. <br>$\vec{A} \cdot \vec{A} = 17$ indicates a maximum similarity.

### 2. Weaviate's Approach:

Distance Metric: Weaviate introduces the ***negative of the dot product*** as a distance metric, where more negative values signal higher similarity:  $-(\vec{A} \cdot \vec{B})$.

###### Example: <br>For vectors $\vec{A}$ and $\vec{B}$, the distance metric in Weaviate, <br>$-(\vec{A} \cdot \vec{B}) = -13$ indicates high similarity. <br>$-(\vec{A} \cdot \vec{A}) = -17$ implies even greater similarity or ***proximity***.

In summary, Weaviate's distance metrics prioritize similarity differently than raw dot product, with lower (more negative) values indicating greater similarity or ***closeness***. 

### 3. Weaviate's Docs:
<img width="719" alt="Screenshot 2024-03-18 at 10 59 36 PM 1" src="https://github.com/weaviate/weaviate-io/assets/92187562/63eb3e21-cc2e-4956-bcb9-d167521fb0fd">
<img width="703" alt="Screenshot 2024-03-18 at 11 01 17 PM" src="https://github.com/weaviate/weaviate-io/assets/92187562/e4188bff-e906-431b-bfe0-d852867f60e4">

Credit: 
[Distance Metric | https://weaviate.io/developers/weaviate/config-refs/distances](https://weaviate.io/developers/weaviate/config-refs/distances)
